src/compiler/scala/tools/nsc/transform/patmat/MatchAnalysis.scala

```diff
@@ -374,10 +374,37 @@ trait MatchApproximation extends TreeAndTypeAnalysis with ScalaLogic with MatchT
         def tru                                               = True
       }
 
+
+      private def isIrrefutabilityProof(sym: Symbol): Boolean = 
+        sym.isMethod && sym.name == TermName("isEmpty") && (sym.asMethod.returnType match {
+          case c: ConstantType => c.value == Constant(false)
```

Comment:
@dwijnand how should I do this to be not wrong? I tried the following, which trips the assertion:

```
      private def isIrrefutabilityProof(sym: Symbol): Boolean = {
        def isConstantFalse1 = sym.asMethod.returnType match {
          case c: ConstantType => c.value == Constant(false)
          case _ => false
        }
        def isConstantFalse2 = sym.asMethod.returnType == ConstantFalse
        sym.isMethod && sym.name == TermName("isEmpty") && {
          assert(isConstantFalse1 == isConstantFalse2, s"symbol $sym has inconsistent return type ${sym.asMethod.returnType}")
          isConstantFalse1
        }
      }
```